### PR TITLE
 patch for debian bullseye

### PR DESCRIPTION
**Pull request by mosfet80**
### Description
This patch is necessary to build moveit2 under debian bullseye

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit2.repos [hunk 1]
```diff
@@ -7,4 +7,56 @@ repositories:
   ros2_control:
     type: git
     url: https://github.com/ros-controls/ros2_control.git
-    version: 2.13.0
+    version: 2.14.0
+  ros-controls/control_msgs:
```
Comment (sjahr) [inline moveit2.repos:11]: On Ubuntu these dependencies get installed with
```
rosdep install -r --from-paths . --ignore-src --rosdistro $ROS_DISTRO -y
```
Doesn't this work on Debian?

File: moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,8 @@ find_package(tf2_ros REQUIRED)
 find_package(orocos_kdl REQUIRED)
 find_package(Boost REQUIRED COMPONENTS)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -std=c++11")
```
Comment (sjahr) [inline moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt:31]: We don't want to use c++11 but a more modern version (Right now c++17). Why is this change necessary for support ob bullseye?

File: moveit2.repos (hunk 1 above)
Comment (mosfet80) [reply moveit2.repos:11]: this list of errors is reported on this distro

~/ws_moveit2$ rosdep install -r --from-paths . --ignore-src --rosdistro $ROS_DISTRO -y
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
moveit_servo: No definition of [launch_param_builder] for OS version []
moveit_resources_prbt_pg70_support: No definition of [xacro] for OS version []
moveit_ros_occupancy_map_monitor: No definition of [octomap] for OS version []
moveit_resources_prbt_support: No definition of [xacro] for OS version []
moveit_core: No definition of [octomap] for OS version []
moveit_kinematics: No definition of [urdfdom] for OS version []
moveit_common: No definition of [backward_ros] for OS version []
moveit_resources_prbt_moveit_config: No definition of [xacro] for OS version []
moveit_resources_fanuc_moveit_config: No definition of [xacro] for OS version []
srdfdom: No definition of [urdfdom_py] for OS version []
geometric_shapes: No definition of [octomap] for OS version []
random_numbers: No definition of [libboost-thread] for OS version []
moveit_configs_utils: No definition of [launch_param_builder] for OS version []
moveit_resources_panda_moveit_config: No definition of [xacro] for OS version []
moveit_resources: No definition of [joint_state_publisher] for OS version []
cv_bridge: No definition of [libboost-python] for OS version []
moveit_hybrid_planning: No definition of [position_controllers] for OS version []
moveit_ros_benchmarks: No definition of [launch_param_builder] for OS version []
Continuing to install resolvable dependencies...
#All required rosdeps installed successfully

Comment (henningkayser) [reply moveit2.repos:11]: It's a bit odd, at least urdfdom should be available, afaik.

> This patch is necessary to build moveit2 under debian bullseye

First off, we currently don't target Debian at all and don't have capacity (or the need) to add or maintain support for it. If you or anyone else are willing to support and maintain it moving forward, we are happy to help you get your changes and fixes in. The moveit2.repos file is already used for maintaining the required workspace dependencies for Ubuntu and should not be changed. Workspace dependencies for Debian or other distros should go into a separate files.
